src/wfpt.pyx

```diff
@@ -70,6 +75,153 @@ def wiener_like(np.ndarray[double, ndim=1] x, double v, double sv, double a, dou
     return sum_logp
 
 
+def wiener_like_rlddm(np.ndarray[double, ndim=1] x,
+                      np.ndarray[long, ndim=1] response,
+                      np.ndarray[double, ndim=1] feedback,
+                      np.ndarray[long, ndim=1] split_by,
+                      double q, double alpha, double dual_alpha, double v, 
+                      double sv, double a, double z, double sz, double t,
+                      double st, double err, int n_st=10, int n_sz=10, bint use_adaptive=1, double simps_err=1e-8,
+                      double p_outlier=0, double w_outlier=0):
+    cdef Py_ssize_t size = x.shape[0]
+    cdef Py_ssize_t i
+    cdef Py_ssize_t s_size
+    cdef int s
+    cdef double p
+    cdef double sum_logp = 0
+    cdef double wp_outlier = w_outlier * p_outlier
+    cdef double alfa
+    cdef np.ndarray[double, ndim=1] qs = np.array([q, q])
+    cdef np.ndarray[double, ndim=1] xs = np.empty(0, dtype=np.double)
+    cdef np.ndarray[double, ndim=1] feedbacks = np.empty(0, dtype=np.double)
+    cdef np.ndarray[long, ndim=1] responses = np.empty(0, dtype=np.long)
+    cdef np.ndarray[long, ndim=1] unique = np.unique(split_by)
+
+    if not p_outlier_in_range(p_outlier):
+        return -np.inf
+
+    # unique represent # of conditions
+    for s in unique:
+        # select trials for current condition, identified by the split_by-array
+        feedbacks = feedback[split_by == s]
+        responses = response[split_by == s]
+        xs = x[split_by == s]
+        s_size = xs.shape[0]
+
+        # don't calculate pdf for first trial but still update q
+        if feedbacks[0] > qs[responses[0]]:
+            alfa = 2.718281828459**(alpha + dual_alpha) / (1 + 2.718281828459**(alpha + dual_alpha))
+        else:
+            alfa = (2.718281828459**alpha) / (1 + 2.718281828459**alpha)
+
+        # qs[1] is upper bound, qs[0] is lower bound. feedbacks is reward
+        # received on current trial.
+        qs[responses[0]] = qs[responses[0]] + \
```

Comment:
So `qs` is just an 1D array with length 2? In that case above it should be `cdef np.ndarray[double, ndim=1] = np.empty(q)` (assuming q == 2).

Reply:
Probably not the best approach, but yeah, qs is 1D array with length 2. q is a single value, so I set both values in qs to be q.

Reply:
Ohhhh, I totally misunderstood what you were doing here, this is fine then.